### PR TITLE
Release `v0.10.0`

### PR DESCRIPTION
**Pull request by angrybayblade**
## Release summary

Version number: v0.10.0

## Release details

Autonomy:

- Adds support for using hardware wallets for minting and managing services on-chain
- Updates the FSM scaffold tool to add the newly scaffolded package to the `packages.json` after scaffolding the skill
- Adds support for `kubernetes` builds on the `from-token` deployments using `--kubernetes` flag 
- Adds support for specifying whether the deployment should run directly or not using `--no-deploy` flag on the `from-deploy` command
- Removes the support for `--force` flag on the autonomy deploy build command
- Adds support for configuring networks in a deployment setup to expose various agent ports 
- Adds a naming convention checker for the FSM specifications. From now on
  - A round name should end with `Round`
  - ABCI app class name should end with `AbciApp`
- Deprecates the usage of the special environment variables from the agent deployment setup
- Adds support for usage of the `flashbots` ledger plugin on the docker images

Packages:
- Deprecates the usage of the special environment variables from the agent configurations
  - In the ABCI skill override don't use `TENDERMINT_URL` and `TENDERMINT_COM_URL` for tendermint parameters
  - In the ABCI connection override don't use `ABCI_HOST` and `ABCI_PORT` for ABCI connection parameters
- Adds a flag called `use_termination` in the configuration to enable or disable the usage of termination
- Adds support to use of the `flashbots` ledger plugin for transactions on the ledger connection

Tests:
- Updates the test to remove the usage of --force flag

Docs:
- Adds the description column to the package list in the docs
- Corrects a code snippet in docs
- Simplified explanation of `what is an agent service`

Chores:
- Update the workflows to use python `3.10.9` instead of `3.10.10` to avoid timeouts

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] I've updated the dependencies versions to the latest, wherever is possible.
- [ ] Lint and unit tests pass locally (please run tests also manually, not only with `tox`)
- [ ] I built the documentation and updated it with the latest changes
- [ ] I've added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `__init__.py` file.
- [ ] I published the latest version on TestPyPI and checked that the following command work:
       ```pip install project-name==<version-number> --index-url https://test.pypi.org/simple --force --no-cache-dir --no-deps```
- [ ] After merging the PR, I'll publish the build also on PyPI. Then, I'll make sure the following
      command will work:
      ```pip install project-name-template==<version_number> --force --no-cache-dir --no-deps```  
- [ ] After merging the PR, I'll tag the repo with `v${VERSION_NUMVER}` (e.g. `v0.1.2`)

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,41 @@
...
+  - ABCI app class name should end with `AbciApp`
+- Deprecates the usage of the special environment variables from the agent deployment setup
+- Adds support for usage of the `flashbots` ledger plugin on the docker images
+
+Packages:
+- Deprecates the usage of the special environment variables from the agent configurations
+  - In the ABCI skill override don't use `TENDERMINT_URL` and `TENDERMINT_COM_URL` for tendermint parameters
+  - In the ABCI connection override don't use `ABCI_HOST` and `ABCI_PORT` for ABCI connection parameters
+- Adds a flag called `use_termination` in the configuration to enable or disable the usage of termination
+- Adds support to use of the `flashbots` ledger plugin for transactions on the ledger connection
```
Comment (Adamantios) [inline HISTORY.md:24]:
```suggestion
- Adds support for the `flashbots` ledger plugin
```

File: docs/upgrading.md [hunk 2]
```diff
@@ -5,6 +5,19 @@ Below we describe the additional manual steps required to upgrade between differ
 
 # Open Autonomy
 
+## `v0.9.1` to `v0.10.0`
+
+Breaking changes
+
+- The services which are currently using termination will need to set `use_termination: True` in their configuration in order to continue using it.
```
Comment (Adamantios) [inline docs/upgrading.md:12]:
```suggestion
- The `AbciAppDB`'s `create()` is responsible for setting the cross-period keys for the new period and converting the corresponding data to the correct format. The skills using the `create()` method now do not need to manually set the data for the next period as this is handled automatically via the cross-period keys (#1827).
- The setup params should not be defined as lists anymore (#1833).
- `observation_interval` has been renamed to `reset_pause_duration` (#1836).
- The services which are currently using termination will need to set `use_termination: True` in their configuration in order to continue using it (#1891).
```

File: HISTORY.md (hunk 1 above)
Comment (0xArdi) [reply HISTORY.md:24]: I agree with this suggestion

File: docs/upgrading.md (hunk 2 above)
Comment (Adamantios) [reply docs/upgrading.md:12]: These will also need to be included in the `HISTORY.md` file.

File: docs/upgrading.md [hunk 3]
```diff
@@ -5,6 +5,29 @@ Below we describe the additional manual steps required to upgrade between differ
...
+  corresponding data to the correct format. The skills using the `create()` method now do not need to manually set the 
+  data for the next period as this is handled automatically via the cross-period keys.
+- The setup parameters should not be defined as lists anymore.
+- `observation_interval` has been renamed to `reset_pause_duration`.
+- The services which are currently using termination will need to set `use_termination: True` 
+  in their configuration in order to continue using it. Otherwise, `use_termination: False` should be used.
+- Before this update the `autonomy mint/service` command groups used to have key file as a required argument, 
+  but it's been made optional since the usage of hardware wallet does not require a key file.
+- Autonomy deploy build does not support the usage of `--force` flag to remove the existing build directories.
+- The service components need to be updated. The service configurations needs to specify the new `deployment` parameter,
```
[the remark this answers is not in the document]
Comment (Adamantios) [reply docs/upgrading.md:22]: This is above:

https://github.com/valory-xyz/open-autonomy/blob/8255c48dd6c4860d68e997fc785e930d15ecc6fc/docs/upgrading.md?plain=1#L17-L18

**Review by 0xArdi — APPROVED** [review]

**Review by dvilelaf — APPROVED** [review]

**Pull request merged**